### PR TITLE
Split out test vs compile

### PR DESCRIPTION
**Pull request by calbach**
Split UI test and compile into separate stages. `ng build` is super spammy and pushes down the test output failures.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -72,9 +72,15 @@ jobs:
...
       - run:
+          name: Run Angular tests
           working_directory: ~/workbench/ui
           command: |
-            npm run codegen && npm run build && npm test -- --no-watch
+            npm run codegen && npm test -- --no-watch
+      - run:
+          name: Test compilation of the Angular app
+          working_directory: ~/workbench/ui
+          command: |
```
Comment (johnstill) [inline .circleci/config.yml]: There's no need for pipes here, is there?  These commands are pretty comfortable on a single line.

Comment (calbach) [reply .circleci/config.yml]: Yep, done.

**Comment by calbach** [comment]
@johnstill I dug more into my suspicions on caching, and I believe they were unfounded. I'm still not sure what caused the discrepancy between local and Circle, but our Circle caching should theoretically not cause issues - unless `npm install` behaves erratically. Regardless of the cache state, we always run `npm install` after restoring the cache - there's just a chance we'll have to do less work with the cache in place.

I also SSH'd into the Circle machine and verified that several relevant module versions matched up with my local client. (and by the way was able to manually repro in SSH via `ng build`).

**Comment by calbach** [comment]
Note: this change is not really functional, it just splits out the logs into different workflows for readability purposes.

I'm not sure that @johnstill is saying the approach wouldn't work, but if so, here is the PR where I originally implemented this approach: https://github.com/all-of-us/workbench/pull/441, the CI run linked there shows that it successfully caught the compilation failure before merge into master. My understanding is that `ng test` does some compilation, but does not compile the entirety of the app (or with same settings).

**Pull request merged**